src/compiler/scanner.ts

```diff
@@ -1556,9 +1556,7 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
                 tokenFlags |= TokenFlags.ContainsInvalidEscape;
                 if (isRegularExpression || shouldEmitInvalidEscapeError) {
                     const code = parseInt(text.substring(start + 1, pos), 8);
-                    if (isRegularExpression !== "annex-b") {

```

Comment:
I thought that too, but there is a `isRegularExpression === true` at the bottom

Reply:
How about swapping it for `"anyUnicodeModeOrNonAnnexB"`?

Reply:
This probably is worth being refactored, since `shouldEmitInvalidEscapeError` does not explain itself but is taken over to mean `anyUnicodeMode` for regular expressions. But this will be yet another PR I guess.

Reply:
Do you use an enum for such a case?